metals/src/main/scala/scala/meta/internal/metals/MtagsResolver.scala

```diff
@@ -143,12 +152,16 @@ object MtagsResolver {
               s"$scalaVersion is no longer supported in the current Metals versions, using the last known supported version $metalsVersion"
             )
           }
-          val jars = fetchResolveType match {
-            case ResolveType.StablePC =>
-              Embedded.downloadScala3PresentationCompiler(scalaVersion)
-            case _ => Embedded.downloadMtags(scalaVersion, metalsVersion)
+          val jarsFuture = Future {
+            fetchResolveType match {
+              case ResolveType.StablePC =>
+                Embedded.downloadScala3PresentationCompiler(scalaVersion)
+              case _ => Embedded.downloadMtags(scalaVersion, metalsVersion)
+            }
           }
 
+          val jars = Await.result(jarsFuture, 60.seconds)
```

Comment:
Should we provide a way to override it for users or this should always be enough?

Reply:
Potentially this should be enough unless the internet is really really bad. In that case it's also probably better to timeout and not hang on downloading.